tests/integration/azure.py

```diff
@@ -606,6 +612,63 @@ def upload_image(self, progress_function = None):
             ).result()
 
             self.logger.info(f"Image {image_file} uploaded as {image_name}")
+
+            gallery_name = self.config['gallery_name']
+            gallery_image_name = self.config['gallery_image_name']
+            gallery_image_version_name = self.config['gallery_image_version_name']
+            self.cclient.galleries.begin_create_or_update(
```

Comment:
What happens with this gallery once the integration tests are completed? Is this gallery then deleted? We should ensure that all test resources are cleaned up if "keep_running" isn't set once the tests have been executed.